docs/050-breaking-changes.rst

```diff
@@ -0,0 +1,85 @@
+hiding new and different behaviors in previous code.
+
+* Signed right shift now uses proper arithmetic shift, i.e. rounding towards
+  negative infinity. Signed and unsigned shift will have dedicated opcodes in
+  Constantinople, and are emulated by Solidity for the moment.
+
+* The ``continue`` statement in a ``do...while`` loop now jumps to the
+  condition, which is the common behavior in such cases. It used to jump to the
+  loop body.
+
+Semantic and Syntactic Changes
+==============================
+
+This section highlights changes that affect syntax and semantics.
+
+* The functions ``.call()`` (and family), ``keccak256()``, ``sha256()``,
+  ``ripemd160()`` now accept only a single ``bytes`` argument. Moreover, the
+  argument is not padded.  TODO: write reason here.  Change every ``.call()``
+  to a ``.call("")`` and every ``.call(signature, a, b, c)`` to use
+  ``.call(abi.encodeWithSignature(signature, a, b, c))`` (the last one only
+  works for value types).  Change every ``keccak256(a, b, c)`` to
+  ``keccak256(abi.encodePacked(a, b, c))``.
+
+* Function ``.call()`` now returns ``(bool, bytes memory)``, since
+  functions called with the opcode ``staticcall`` might need to return data.
+  Change ``bool success = otherContract.call("f")`` to ``(bool success, bytes
+  memory data) = otherContract.call("f")``.
+
+* Solidity now implements C99-style scoping rules for function local variables,
+  that is, a local variable declared in the function is only valid if declared
+  in the same or parenting scope. Declare function variables before using them
+  in the same or deeper scope.
+
+Explicitness requirements
+=========================
+
+This section lists changes where the code now needs to be more explicit.
+
+* Explicit function visibility is now mandatory.  Add ``public`` to every
+  function and constructor, and ``external`` to every fallback or interface
+  function that does not specify its visibility already.
+
+* Explicit data location for all variables of struct, array or mapping types is
+  now mandatory. This is also applied to function parameters and return
+  variables.  For example, change ``uint[] x = m_x`` to ``uint[] storage x =
+  m_x``, and ``function f(uint[][] x)`` to ``function f(uint[][] memory x)``
+  where ``memory`` is the data location and might be replaced by ``storage`` or
+  ``calldata`` accordingly.  Note that ``external`` functions require
+  parameters with a data location of ``calldata``.
+
+* Variables of contract type do not contain ``address`` members anymore.
+  Therefore, it is now necessary to explicitly convert values of contract type
+  to addresses before using an ``address`` member.  Example: if ``c`` is a
+  contract, change ``c.transfer(...)`` to ``address(c).transfer(...)``.
+
+* The ``address`` type  was split into ``address`` and ``address payable``.
+  Function ``transfer`` cannot be called by an ``address``, only by an
```

Comment:
"called by" sounds a little weird to me. What about "Only `address payable` provides the `transfer` function"?